internal/observability/graph.go

```diff
@@ -230,6 +251,141 @@ func newEdge(logger *zap.Logger, meter *metrics.Scope, req *transport.Request, d
 	if err != nil {
 		logger.Error("Failed to create server failure latency distribution.", zap.Error(err))
 	}
+
+	// streaming requests
+	var streaming *streamEdge
+	if rpcType == transport.Streaming {
+		// sends
+		sends, err := meter.Counter(metrics.Spec{
+			Name:      "stream_sends",
+			Help:      "Total number of streaming messages sent.",
+			ConstTags: tags,
+		})
+		if err != nil {
+			logger.Error("Failed to create streaming sends counter.", zap.Error(err))
+		}
+		sendSuccesses, err := meter.Counter(metrics.Spec{
+			Name:      "stream_send_successes",
+			Help:      "Number of successful streaming messages sent.",
+			ConstTags: tags,
+		})
+		if err != nil {
+			logger.Error("Failed to create streaming sends successes counter.", zap.Error(err))
+		}
+		sendFailures, err := meter.CounterVector(metrics.Spec{
+			Name:      "stream_send_failures",
+			Help:      "Number streaming messages that failed to send.",
+			ConstTags: tags,
+			VarTags:   []string{_error},
+		})
+		if err != nil {
+			logger.Error("Failed to create streaming sends failure counter.", zap.Error(err))
+		}
+
+		// receives
+		receives, err := meter.Counter(metrics.Spec{
+			Name:      "stream_receives",
+			Help:      "Total number of streaming messages recevied.",
+			ConstTags: tags,
+		})
+		if err != nil {
+			logger.Error("Failed to create streaming receives counter.", zap.Error(err))
+		}
+		receiveSuccesses, err := meter.Counter(metrics.Spec{
+			Name:      "stream_receives_successes",
+			Help:      "Number of successful streaming messages received.",
+			ConstTags: tags,
+		})
+		if err != nil {
+			logger.Error("Failed to create streaming receives successes counter.", zap.Error(err))
+		}
+		receiveFailures, err := meter.CounterVector(metrics.Spec{
+			Name:      "stream_receives_failures",
+			Help:      "Number streaming messages failed to be recieved.",
+			ConstTags: tags,
+			VarTags:   []string{_error},
+		})
+		if err != nil {
+			logger.Error("Failed to create streaming receives failure counter.", zap.Error(err))
+		}
+
+		// histograms
+		streamDurations, err := meter.Histogram(metrics.HistogramSpec{
+			Spec: metrics.Spec{
+				Name:      "stream_duration_ms",
+				Help:      "Latency distribution of total stream duration.",
+				ConstTags: tags,
+			},
+			Unit:    time.Millisecond,
+			Buckets: _bucketsMs,
+		})
+		if err != nil {
+			logger.Error("Failed to create stream duration histogram.", zap.Error(err))
+		}
+		sendLatencies, err := meter.Histogram(metrics.HistogramSpec{
+			Spec: metrics.Spec{
+				Name:      "stream_send_success_latency_ms",
+				Help:      "Latency distribution of successful streaming message sends.",
```

Comment:
not clear from the help what this actually measures. is this the time for a send message to complete? does that give us any signal?